module/programs/helper.js

```diff
@@ -150,6 +197,827 @@ module.exports = class ProgramsHelper {
+      try {
+        let programData = await programsQueries.programDocuments({
+          _id: programId,
+        });
+
+        if (!(programData.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
+          });
+        }
+
+        return resolve({
+          message: messageConstants.apiResponses.PROGRAMS_FETCHED,
+          success: true,
+          data: programData[0],
+        });
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : httpStatusCode["internal_server_error"].status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+   
+
+  /**
+   * Program join.
+   * @method
+   * @name join
+   * @param {String} programId - Program Id.
+   * @param {Object} data - body data (can include isResourse flag && userRoleInformation).
+   * @param {String} userId - Logged in user id.
+   * @param {String} userToken - User token.
+   * @param {String} [appName = ""] - App Name.
+   * @param {String} [appVersion = ""] - App Version.
+   * @param {Boolean} callConsetAPIOnBehalfOfUser - required to call consent api or not
+   * @returns {Object} - Details of the program join.
+   */
+
+  static join(
+    programId,
+    data,
+    userId,
+    userToken,
+    appName = "",
+    appVersion = "",
+    callConsetAPIOnBehalfOfUser = false
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let pushProgramUsersDetailsToKafka = false;
+        //Using programId fetch program details. Also checking the program status in the query.
+        let programData = await programsQueries.programDocuments(
+          {
+            _id: programId,
+            status: messageConstants.common.ACTIVE_STATUS,
+            isDeleted: false,
+          },
+          ["name", "externalId", "requestForPIIConsent", "rootOrganisations"]
+        );
+
+        if (!(programData.length > 0)) {
+          throw {
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
+          };
+        }
+        let programUsersData = {};
+        let update = {};
+
+        // check if user already joined for program or not
+        const programUsersDetails =
+          await programUsersQueries.programUsersDocument(
+            {
+              userId: userId,
+              programId: programId,
+            },
+            ["_id", "consentShared"]
+          );
+        // if user not joined for program. we have add more key values to programUsersData
+        if (!(programUsersDetails.length > 0)) {
+          // Fetch user profile information by calling elevate-user  read api through interface service.
+          // !Important check specific fields of userProfile.
+          let userProfile = await userService.profile(userToken, userId);
```

Comment:
@praveenKDass if you are calling user service. user/read api. that will not need token. it requires internal access token

Reply:
Changed passed internal-access-token @VISHNUDAS-tunerlabs